File: src/dataset/Java/pangram_exercism-java-blogscot/gemini/GeminiFlash_pangram_v3.java
class PangramChecker {

  boolean isPangram(String input) {
    if (input == null || input.length() < 26) {
      return false;
    }

    long seenLetters = 0L;
    final long ALL_LETTERS_MASK = (1L << 26) - 1;

    for (int i = 0; i < input.length(); i++) {
      char c = input.charAt(i);

      if (c >= 'a' && c <= 'z') {
        seenLetters |= (1L << (c - 'a'));
      } else if (c >= 'A' && c <= 'Z') {
        seenLetters |= (1L << (c - 'A'));
      }

      if (seenLetters == ALL_LETTERS_MASK) {
        return true;
      }
    }

    return seenLetters == ALL_LETTERS_MASK;
  }

}

File: src/dataset/go/pangram_Exercism_go_drapala/gemini/GeminiFlash_pangram_v3.go
package pangram

import (
	"unicode"
)

func IsPangram(input string) bool {
	var seenLetters uint32
	const allLettersMask uint32 = (1 << 26) - 1 // A bitmask where all 26 bits (0-25) are set, representing 'a' through 'z'

	for _, r := range input {
		lcR := unicode.ToLower(r) // Convert rune to its lowercase equivalent

		// Check if the character is an English lowercase letter
		if lcR >= 'a' && lcR <= 'z' {
			// Calculate the bit position for this letter (0 for 'a', 1 for 'b', ..., 25 for 'z')
			bitPos := lcR - 'a'
			// Set the corresponding bit in our seenLetters bitmask
			seenLetters |= (1 << bitPos)

			// Early exit: If all 26 bits are set, we've found all unique letters
			if seenLetters == allLettersMask {
				return true
			}
		}
	}

	// After iterating through the entire string, check if all 26 bits are set
	return seenLetters == allLettersMask
}